Tangem/App/Factories/WalletModels/CommonWalletModelsFactory.swift

```diff
@@ -31,23 +31,32 @@ struct CommonWalletModelsFactory {
             return FakeTransactionHistoryService(blockchain: tokenItem.blockchain, address: address)
         }
 
-        let factory = TransactionHistoryFactoryProvider().factory
-        guard let provider = factory.makeProvider(for: tokenItem.blockchain) else {
-            return nil
-        }
-
         if addresses.count == 1, let address = addresses.first {
+            let factory = TransactionHistoryFactoryProvider().factory
+
+            guard let provider = factory.makeProvider(for: tokenItem.blockchain) else {
+                return nil
+            }
+
             return CommonTransactionHistoryService(
                 tokenItem: tokenItem,
                 address: address,
                 transactionHistoryProvider: provider
             )
         }
 
+        let multiAddressProviders: [String: TransactionHistoryProvider?] = [:]
+
+        _ = addresses.reduce(into: multiAddressProviders) { partialResult, address in
```

Comment:
что ж у все с `reduce` проблемы ))

```suggestion
        let multiAddressProviders: [String: TransactionHistoryProvider?] = addresses.reduce(into: [:]) { partialResult, address in
```

Reply:
Проверю , может разные варианты пробовал осталось , спс